server/master/src/main/java/org/apache/accumulo/master/TabletGroupWatcher.java

```diff
@@ -614,7 +615,8 @@ private void deleteTablets(MergeInfo info) throws AccumuloException {
       for (Entry<Key,Value> entry : scanner) {
         Key key = entry.getKey();
         if (key.compareColumnFamily(DataFileColumnFamily.NAME) == 0) {
-          datafiles.add(new FileRef(this.master.fs, key));
+          datafiles
+              .add(new FileRef(TabletFileUtil.validate(key.getColumnQualifierData().toString())));
```

Comment:
With a method like the following, could possibly do `datafiles.add(new FileRef(TabletFile.from(key)));`.  Not completely sure about this, its just something I thought of while looking at this code.

```java
class TableFile {
 public static TableFile from(Key k){
   switch(k.getColumnFamilyData().toString()) {
     case SCAN:
     case DATA:
         return new TableFile(k.getColumnQualifierData().toString());
   }
 }
}
```